src/block_ffm.rs

```diff
@@ -262,8 +315,32 @@ impl<L: OptimizerTrait + 'static> BlockTrait for BlockFFM<L> {
 
                             for z in 0..ffm_fields_count_as_usize {
                                 let general_gradient = myslice.get_unchecked(contra_offset + z);
+                                let general_gradient_simd = f32x4::splat(*general_gradient);
 
-                                for k in 0.. ffmk_as_usize {
+                                for k in (0..ffmk_end).step_by(STEP) {
+                                    let feature_value_simd = f32x4::from_slice(local_data_ffm_values.get_unchecked(local_index..local_index + STEP));
+                                    let gradient_simd = general_gradient_simd * feature_value_simd;
+                                    let gradient = gradient_simd.as_array();
+
+                                    let update = self.optimizer_ffm.calculate_update(gradient[0],
```

Comment:
Why unrolling to 4? This feels tied to `k` hyperparameter, with possible pitfalls for different values of it?